Loop/Managers/DeviceDataManager.swift

```diff
@@ -1421,8 +1421,39 @@ extension DeviceDataManager {
                 }
             }
         }
-        // Wait up to 25 seconds for uploads triggered by these commands to finish
-        let _ = remoteDataServicesManager.waitForUploadsToFinish(timeout: .now() + TimeInterval(25))
+    }
+    
```

Comment:
The pattern for these 3 methods is similar:

1. Perform the treatment action.
2. Call `triggerBackgroundUpload` to force the upload to happen now so we have something to wait on. 

Without triggering the upload explicitly, we can't rely on the groupTasks having been started yet by this time.